hazelcast/errors.py

```diff
@@ -0,0 +1,646 @@
+    _UTF_DATA_FORMAT: UTFDataFormatError,
+    _UNSUPPORTED_OPERATION: UnsupportedOperationError,
+    _WRONG_TARGET: WrongTargetError,
+    _XA: XAError,
+    _ACCESS_CONTROL: AccessControlError,
+    _LOGIN: LoginError,
+    _UNSUPPORTED_CALLBACK: UnsupportedCallbackError,
+    _NO_DATA_MEMBER: NoDataMemberInClusterError,
+    _REPLICATED_MAP_CANT_BE_CREATED: ReplicatedMapCantBeCreatedOnLiteMemberError,
+    _MAX_MESSAGE_SIZE_EXCEEDED: MaxMessageSizeExceededError,
+    _WAN_REPLICATION_QUEUE_FULL: WANReplicationQueueFullError,
+    _ASSERTION_ERROR: HazelcastAssertionError,
+    _OUT_OF_MEMORY_ERROR: OutOfMemoryError,
+    _STACK_OVERFLOW_ERROR: StackOverflowError,
+    _NATIVE_OUT_OF_MEMORY_ERROR: NativeOutOfMemoryError,
+    _SERVICE_NOT_FOUND: ServiceNotFoundError,
+    _STALE_TASK_ID: StaleTaskIdError,
+    _DUPLICATE_TASK: DuplicateTaskError,
+    _STALE_TASK: StaleTaskError,
+    _LOCAL_MEMBER_RESET: LocalMemberResetError,
+    _INDETERMINATE_OPERATION_STATE: IndeterminateOperationStateError,
+    _FLAKE_ID_NODE_ID_OUT_OF_RANGE_EXCEPTION: NodeIdOutOfRangeError,
+    _TARGET_NOT_REPLICA_EXCEPTION: TargetNotReplicaError,
+    _MUTATION_DISALLOWED_EXCEPTION: MutationDisallowedError,
+    _CONSISTENCY_LOST_EXCEPTION: ConsistencyLostError,
+    _VERSION_MISMATCH_EXCEPTION: VersionMismatchError,
+    _NO_SUCH_METHOD_ERROR: NoSuchMethodError,
+    _NO_SUCH_METHOD_EXCEPTION: NoSuchMethodException,
+    _NO_SUCH_FIELD_ERROR: NoSuchFieldError,
+    _NO_SUCH_FIELD_EXCEPTION: NoSuchFieldException,
+    _NO_CLASS_DEF_FOUND_ERROR: NoClassDefFoundError,
+}
+
+from hazelcast.protocol.builtin import ListMultiFrameCodec
+from hazelcast.protocol.codec.custom.error_holder_codec import ErrorHolderCodec
+
+
+class _ErrorsCodec(object):
+    @staticmethod
+    def decode(msg):
+        msg.next_frame()
+        return ListMultiFrameCodec.decode(msg, ErrorHolderCodec.decode)
+
+
+def create_error_from_message(error_message):
+    """
+    Creates an exception with given error codec.
+
+    :param error_message: (ClientMessage), error message which includes the class name, message and exception trace.
+    :return: (Exception), the created exception.
+    """
+    error_holders = _ErrorsCodec.decode(error_message)
+    return _create_error(error_holders, 0)
+
+
+def _create_error(error_holders, idx):
+    if idx == len(error_holders):
+        return None
+
+    error_holder = error_holders[idx]
+    error_class = _ERROR_CODE_TO_ERROR.get(error_holder.error_code, None)
+
+    stack_trace = "\n".join(
+        ["\tat %s.%s(%s:%s)" % (x.class_name, x.method_name, x.file_name, x.line_number) for x in
+         error_holder.stack_trace_elements])
+    message = "Exception from server: %s: %s\n %s" % (error_holder.class_name, error_holder.message, stack_trace)
+    if error_class:
+        return error_class(message, _create_error(error_holders, idx + 1))
+    else:
+        return UndefinedErrorCodeError(message, error_holder.class_name)
```

Comment:
I have added cause to `UndefinedErrorCodeError` on java. You can do the same here as well.